### ISSUE TITLE
DNSDIST restarts when DoH query made

### ISSUE THREAD

**Issue by allahshukur-ahmadzada** [PowerDNS/pdns#12645]
When running DoH query agains dnsdist which kdig for first query I get result but  right after it if I do second query I get error and when I check dnsdist logs I see its restartted with below logs

```
Mar 13 14:21:49 aws-nameserver dnsdist[2026984]: Got an error in DOH question thread while parsing a query from 162.219.4.28:52452, id 0: Attempt to print an unset dnsname
Mar 13 14:21:49 aws-nameserver dnsdist[2026984]: dnsdist: /build/h2o-tgUKZx/h2o-2.2.5+dfsg2/lib/http2/stream.c:339: finalostream_send: Assertion `stream->_data.size == 0' failed.
Mar 13 14:21:51 aws-nameserver dnsdist[2027066]: Configuration '/etc/dnsdist/dnsdist.conf' OK!
Mar 13 14:21:51 aws-nameserver dnsdist[2027066]: Configuration '/etc/dnsdist/dnsdist.conf' OK!
Mar 13 14:21:51 aws-nameserver dnsdist[2027067]: Read configuration from '/etc/dnsdist/dnsdist.conf'
Mar 13 14:21:51 aws-nameserver dnsdist[2027067]: Loading default TLS provider openssl
```
kdig query:

> kdig -d @IP +https=hostname example.com

dnsdist version:
```
dnsdist 1.7.3 (Lua 5.1.4 [LuaJIT 2.1.0-beta3])
Enabled features: cdb dns-over-tls(gnutls openssl) dns-over-https(DOH) dnscrypt ebpf fstrm ipcipher libsodium lmdb outgoing-dns-over-https(nghttp2) protobuf re2 recvmmsg/sendmmsg snmp systemd
```

dnsdist config which forwards qeuries to PowerDNS recursor:
```
addTLSLocal('172.31.4.54', '/etc/dnsdist/certs/cert.pem', '/etc/dnsdist/certs/key.pem')
addDOHLocal('172.31.4.54', '/etc/dnsdist/certs/cert.pem', '/etc/dnsdist/certs/key.pem')
addACL("0.0.0.0/0")
newServer({
 address="172.31.4.54:53"
})
```
OS:

> Ubuntu 20.04.5 LTS

**Comment by Habbie** [comment]
Looking at the code, the last message is a `vinfolog`, so I asked to try without `-v` - the crash is gone.

**Comment by rgacogne** [comment]
OK, so we are likely really trying to print an unset `DNSName`. Unfortunately I can't reproduce the issue with that configuration and `kdig` command, but I'll try to trigger this in another way.

**Comment by rgacogne** [comment]
This issue should be fixed by #12646, and doesn't exist in the main branch which will become 1.8.
